component/async/kafka/simple/duration_client.go

```diff
@@ -0,0 +1,130 @@
+package simple
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/beatlabs/patron/component/async/kafka"
+	"github.com/beatlabs/patron/log"
+)
+
+type durationClient struct {
+	client durationKafkaClientAPI
+}
+
+func newDurationClient(client durationKafkaClientAPI) (durationClient, error) {
+	if client == nil {
+		return durationClient{}, errors.New("empty client api")
+	}
+	return durationClient{client: client}, nil
+}
+
+func (d durationClient) getTimeBasedOffsetsPerPartition(ctx context.Context, topic string, since time.Time, timeExtractor kafka.TimeExtractor) (map[int32]int64, error) {
+	partitionIDs, err := d.client.getPartitionIDs(topic)
+	if err != nil {
+		return nil, err
+	}
+
+	responseCh := make(chan partitionOffsetResponse, 1)
```

Comment:
in order to not block at all
```suggestion
	responseCh := make(chan partitionOffsetResponse, len(partitionIDs))
```